### ISSUE TITLE
Ambiguous path is not allowed in tagged template

### ISSUE THREAD

**Issue by amk221** [ember-template-lint/ember-template-lint#463]
When running with the new `no-implicit-this` rule, my test files are being picked up, for example:

```javascript
test('something', async function(assert) {
  this.set('thing', {
    id: 123
  });

  await render(hbs`{{my-thing thing=thing}}`);
});
```

`Ambiguous path 'thing' is not allowed. Use '@thing' `

But the test fails if I change it to that.

It's only complaining about a small selection of tests, despite there being many other similar tests
(I'm not sure what it special about the few).

**Comment by rwjblue** [comment]
Hmm, there are a few different pieces here I think.

* to have the inline test templates checked I think you must be using `eslint-plugin-hbs`. `eslint-plugin-hbs` is hard coded to use the 0.8.x version of `ember-template-lint` ([see here](https://github.com/psbanka/eslint-plugin-hbs/blob/b3b6447339ad2081f9d8cda2667e3dd8daaf61f9/package.json#L18)) which makes interop quite a bit trickier.  I created https://github.com/psbanka/eslint-plugin-hbs/issues/25 to chat through that.
* I don't think that (in the snippet you gave) you should swap to `@thing`, it should instead be `this.thing`. Do the tests pass if you do it that way?
* I am not sure why only some tests have the issue, can you test by copying one of the ones that work into a file of one that doesn't, and see if it still complains?

Thanks for testing and reporting!

**Comment by amk221** [comment]
* Adding `this.` does silence the tests as expected
* I _think_ the ones that are not erroring as expected happen to have a dash prefixing their name, e.g. `my-component/-foo/bar`

**Comment by amk221** [comment]
I've also found inconsistencies... sometimes I get notified about:

```
hbs`{{my-thing click=(action doSomthing)}}`
```

and othertimes, I get prompted to add `this.`

```
hbs`{{my-thing click=(action this.doSomthing)}}`
```

**Comment by rwjblue** [comment]
> it seems print-failing and pending does not take account of the test files (with tagged templates)?

Definitely something we should address. First we need that eslint-plugin-hbs a valid module for the file, then we should we arbitrarily assume `.hbs` vs `.js`.

**Comment by amk221** [comment]
Ok thanks, I'll track that issue you linked to!

What about the other point I mentioned? Would I be right in thinking, that to create a failing test I should add: 

`{{helper}}`

Here: https://github.com/ember-template-lint/ember-template-lint/blob/master/test/unit/rules/lint-no-implicit-this-test.js#L23 ?
